pandas/io/formats/style.py

```diff
@@ -575,75 +564,99 @@ def _translate(self):
 
         return d
 
-    def format(self, formatter, subset=None, na_rep: Optional[str] = None) -> Styler:
+    def format(
+        self,
+        formatter: Optional[
+            Union[Dict[Any, Optional[Union[str, Callable]]], str, Callable]
+        ] = None,
+        subset=None,
+        na_rep: Optional[str] = None,
+    ) -> Styler:
         """
         Format the text display value of cells.
 
         Parameters
         ----------
         formatter : str, callable, dict or None
-            If ``formatter`` is None, the default formatter is used.
+            Format specification to use for displaying values. If ``None``, the default
+            formatter is used. If ``dict``, keys should corresponcd to column names,
+            and values should be string or callable.
         subset : IndexSlice
             An argument to ``DataFrame.loc`` that restricts which elements
             ``formatter`` is applied to.
         na_rep : str, optional
-            Representation for missing values.
-            If ``na_rep`` is None, no special formatting is applied.
+            Representation for missing values. If ``None``, will revert to using
+            ``Styler.na_rep``
```

Comment:
hmm why are you adding this special function? we don't do this anywhere else

Reply:
I think your comment about the fact that nowhere else is there a `set_na_rep` method that lends itself to the answer here. It is possible for there to be a solution where both functions (that currently exists) operate consistently, as opposed to somewhat inconsistently and inconveniently at present

Reply:
I would add that this might not even be interpreted as a change. `no special formatting applied` might be expected to be that `Styler.na_rep`` is used as default, it is ambiguous. I was just clarifying what the bug fix did here.